src/core_plugins/elasticsearch/lib/kibana_index_mappings.js

```diff
@@ -1,16 +1,36 @@
-export const mappings = {
-  config: {
-    properties: {
-      buildNum: {
-        type: 'keyword'
-      }
-    }
-  },
-  server: {
-    properties: {
-      uuid: {
-        type: 'keyword'
+import _ from 'lodash';
+
+class KibanaMappings {
+  constructor(plugin) {
+    this._plugin = plugin;
+    this._defaultMappings = {
+      '_default_': {
+        'dynamic': 'strict'
+      },
+      config: {
+        dynamic: true,
```

Comment:
It doesn't need to be, we may even be able to set it to enabled:false.  I wanted to do that but I thought it could be very breaking for a point release.

There's no fallback for it now however like we have with the other mappings.  The advanced settings API lets you arbitrarily register new settings without mappings, so it would be more to handle.  The frontend would probably have to support a dynamic-like behavior until the major release.